bunch of datum mats stuff mostly related to coins

Fikou opened the pull request:
## About The Pull Request

adamantine and mythril are now datum mats (mythril still admin only), adamantine has 1.5 strength buff to whatever is made out of it, mythril adds rpg loot elements to whatever is made out of it
you can now put plastic, adamantine and mythril in the autolathe and coin mint
you can now put titanium in coin mint
switches a bunch of stuff in cargo exports, including mythril and plastic being worth cash money
coin code has been changed to datum mats, the custom sprites have been lost except for antag token

## Why It's Good For The Game

allows for much more freedom with your coins, fun things in autolathe, and makeable adamantine coins

## Changelog
:cl:
add: You can now put a bunch more mats in the coin mint and autolathe
add: Adamantine and Mythril are now materials (Adamantine still only from xenobio, Mythril still only from badminnery). Adamantine boosts an item's force by 1.5, Mythril gives an item RPG statistics.
del: most custom sprites for coins have been lost
tweak: You can now sell plastic in cargo for actual cash money
fix: you can no longer make nonexistant bamboo walls
/:cl:

Fikou committed:
fixy wixy

Qustinnus commented:
thanks, I was sure to miss a few places in the rework

Fikou committed:
adds a titanium coin

Fikou commented:
huh nevermind, it was that no titanium coins existed, i made one but it uses silver coin sprite, im gonna make new ones soon

Qustinnus commented:
@Fikou no don't, make a greyscale coin and use material datums.

Fikou committed:
removes coin_type

Fikou committed:
removes coin_type

Fikou committed:
does things

Fikou committed:
big coin changes

Fikou committed:
more mats

Fikou committed:
hmm

Fikou committed:
epic

Fikou committed:
epic

Fikou committed:
rpg magic

Fikou committed:
really magical how dumb i am

Fikou committed:
a

Fikou committed:
FUCK

Fikou committed:
MOAR COINS

Fikou committed:
plastock

Fikou committed:
h

Fikou committed:
cool mats in autolathe

Iatots commented:
Does plastatium also have a material datum? What about alien alloy?

Fikou committed:
h

Fikou committed:
h

Qustinnus commented:
they dont because alloys are handled a bit weirdly, they are not really counted as their own material but instead count as 2 seperate materials (it was always like this) so someone needs to change that. not a super hard change but someone has to do it at some point

Qustinnus commented on code/modules/mining/ores_coins.dm:
```diff
@@ -304,17 +304,29 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 	icon = 'icons/obj/economy.dmi'
 	name = "coin"
-	icon_state = "coin__heads"
+	icon_state = "coin_[customsprite]_[coinflip]"
```
this means the default icon state is flipping

Fikou replied:
why, coinflip is a var that picks between two sides of the coin
coin_[customsprite]_flip is the flipping animation

Qustinnus replied:
ooh disregard that then

Fikou committed:
oh god

Fikou committed:
purease work

Fikou committed:
henk

Fikou committed:
a

Fikou committed:
grug

Fikou committed:
GRUGHAUOHGADHA

Fikou committed:
yaaatatata

Fikou committed:
bruh

Fikou committed:
a

Fikou committed:
bruh

Fikou committed:
rip

Fikou committed:
b

Fikou committed:
nicee

Fikou committed:
i think im finished

Fikou committed:
nerf

Fikou committed:
meh

Fikou committed:
please work

Fikou commented:
qustinnus took me on a journey where i changed a ton of shit in datum minerals and added new ones
i just wanted to make titanium coins to fuel my powergaming

Fikou committed:
hnnnnnnnnnn

Qustinnus commented on code/modules/mining/ores_coins.dm:
```diff
@@ -345,94 +356,49 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 
 /obj/item/coin/antagtoken
 	name = "antag token"
-	icon_state = "coin_valid_valid"
```
invalid icon_state by default is causing the unit tests to fail

Fikou replied:
yes im gonna fix that when i get to the dmi making

ExcessiveUseOfCobblestone commented:
VERIFIED

ExcessiveUseOfCobblestone commented on code/modules/mining/ores_coins.dm:
```diff
@@ -345,94 +356,49 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
+
+/obj/item/coin/iron
 
 /obj/item/coin/attackby(obj/item/W, mob/user, params)
```
why is there stuff about subtypes sandwiched between base coin procs? they should all be together :(

Fikou replied:
the reason is that i didnt want to make map edits, if i really need to, ill remove them

ExcessiveUseOfCobblestone replied:
how would the placement of the items within the file touch map edits?

Qustinnus replied:
he's misunderstanding. @Fikou you need to move the subtype somewhere nicer, also you can optionally just do find and replace in the _maps folder by replacing /obj/item/coin/iron with /obj/item/coin and it will work fine

Fikou replied:
oh ok put them next to the normal coin

Fikou committed:
grug grug

ExcessiveUseOfCobblestone commented on code/modules/mining/ores_coins.dm:
```diff
@@ -304,18 +304,71 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 /obj/item/coin
```
this file needs to go like this

```DM
/obj/item/coin
    stuff
    var/stuff

/obj/item/coin/proc1
    stuff

/obj/item/coin/proc2
    stuff

/obj/item/coin/subtype
    stuff
    var/stuff

/obj/item/coin/subtype2
    stuff

ExcessiveUseOfCobblestone replied:
put all `/obj/item/coin` procs above any subtypes

Fikou replied:
ok

Fikou committed:
unga unga bunga bunga

ExcessiveUseOfCobblestone commented:
well you got the file organization right at least

Fikou committed:
fuck goofconomy

Fikou commented:
where my nigga icondiffbot

Fikou committed:
h

Qustinnus commented on code/modules/mining/mint.dm:
```diff
@@ -102,6 +103,7 @@
 	if(T)
 		var/obj/item/O = new P(src)
 		var/obj/item/storage/bag/money/M = locate(/obj/item/storage/bag/money, T)
+		O.set_custom_materials()
```
Don't forget to actually set the args for this proc

Fikou replied:
dose this work

Fikou committed:
shgahahad

Fikou committed:
hewwo

Fikou committed:
h

Qustinnus commented on code/modules/mining/ores_coins.dm:
```diff
@@ -304,18 +304,31 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
+	pixel_y = rand(0,8)-8
+
+/obj/item/coin/set_custom_materials()
+	. = ..()
```
add value = 0 to prevent the value from going up indefinitely when you change materials

Fikou committed:
higgas

Fikou committed:
stinky smelly

Fikou committed:
based reddit cringe

Fikou committed:
hoes mad

Fikou committed:
woopsie

Fikou committed:
retat

Fikou committed:
Update ores_coins.dm

Fikou committed:
yatta

Fikou committed:
h

ExcessiveUseOfCobblestone commented:
Verified

Fikou committed:
israel

Fikou commented:
> Verified

verifies ^w^
i think it should be ready for review now

Commit:
FIXES MY LIFE

Commit:
yes

Fikou committed:
Merge pull request #4 from Qustinnus/whatthefuck

yeet

Qustinnus commented:
![image](https://user-images.githubusercontent.com/20987591/64738536-bcb8a300-d4ef-11e9-857a-c6fb72eaf7b9.png)

Krysonism commented:
Make new custom sprites instead of removing the old ones.

Qustinnus commented:
I can just read the custom sprites in a pr after this but I didn't wanna bother going through how to set that up with fikou

Fikou committed:
a

Fikou committed:
h

Fikou committed:
h

Fikou committed:
hh

Fikou committed:
hhhhhhhhhhhhhh

Fikou committed:
zoomin ayy

Fikou committed:
Merge branch 'master' into patch-51

Fikou committed:
oy vey

Fikou committed:
aaaaaaaaa

Fikou committed:
a

Fikou committed:
aaaaaaaaaaaa

Fikou committed:
hhhhhh

Qustinnus commented on code/game/machinery/autolathe.dm:
```diff
@@ -1,444 +1,447 @@
+		/datum/material/plastic, 
+		/datum/material/adamantine,
+		/datum/material/mythril,
+    /datum/material/runite
```
nice consistency

Fikou committed:
f

Fikou committed:
cry

Fikou committed:
yeet

Fikou committed:
yee

Fikou committed:
yeet

Fikou committed:
hiie

Fikou committed:
h

Fikou committed:
Merge branch 'master' into patch-51

Fikou committed:
aaaaa

Fikou commented:
I FINISHED EVERYTHING WORKS FUCK YEAH

ninjanomnom commented on code/game/machinery/autolathe.dm:
```diff
@@ -1,444 +1,447 @@
-#define AUTOLATHE_MAIN_MENU       1
```
You seem to have messed up line endings here

Fikou replied:
wahooooooooooooooooooooooooooooooooooooooooooo

Fikou committed:
a

Fikou committed:
fuck

Fikou committed:
Update autolathe.dm

Fikou committed:
Update autolathe.dm

Fikou committed:
Update autolathe.dm

Fikou committed:
Update autolathe.dm

Fikou committed:
Update autolathe.dm

Fikou committed:
Merge branch 'master' into patch-51

Fikou committed:
Merge branch 'master' into patch-51

Fikou committed:
Merge branch 'patch-51' of https://github.com/Fikou/tgstation into patch-51

Fikou committed:
f

Fikou committed:
purease

Fikou committed:
Update autolathe.dm

Fikou committed:
brooooo......

Fikou committed:
f

Qustinnus commented:
hell yeah gamer time

optimumtact commented:
Good job you introduced an infinite loop in slot machine payouts.

Qustinnus commented:
how can it be a fuckup of the year 2020 if it happend in 2019

Fikou commented:
this is a pr from 2019 hello?

Fikou commented:
also i blame floyd he told me to not make this pr adding a single new coin type
